### PR TITLE
feat: wrap/unwrap AES files keys

### PR DESCRIPTION
**Pull request by paultranvan**
* Wrap the AES file key at upload time 
* Unwrap the AES file key at download time

### CODE REVIEW

File: src/drive/web/modules/encryption/data.js [hunk 1]
```diff
@@ -37,12 +36,13 @@ export const decryptData = async (key, data, { algorithm, iv } = {}) => {
...
 
-export const createDecryptedFileURL = async file => {
-  // prepare decryption
+export const createDecryptedFileURL = async (vaultKey, file) => {
+  // Prepare key + binary decryption
   const encryption = file.metadata.encryption
-  const iv = decodeArrayBuffer(encryption.iv)
-  // TODO the key should be encrypted
-  const key = await importKey('jwk', encryption.key)
+  const wrappedKey = decodeArrayBuffer(encryption.key.encrypted_key)
```
Comment (francoisWeber) [inline src/drive/web/modules/encryption/data.js:42]: Why a snake case for `encrypted_key` here ?

**Review by francoisWeber — COMMENTED** [review]
Besides the snake case it looks good to me!

File: src/drive/web/modules/encryption/data.js (hunk 1 above)
Comment (paultranvan) [reply src/drive/web/modules/encryption/data.js:42]: See the `encrypted_key` param in https://tools.ietf.org/html/rfc7516#section-3.2

Comment (francoisWeber) [reply src/drive/web/modules/encryption/data.js:42]: My bad, you already told me once

Comment (paultranvan) [reply src/drive/web/modules/encryption/data.js:42]: I forgive you because of your insights on the snake/camel cases

**Pull request merged**